### ISSUE TITLE
Simple multithreaded app crashed

### ISSUE THREAD

**Issue by h66m9d** [shushudu/OPCDAClientSync#1]
I am trying to use this code in  multithread simple app. but it doesn't work.
`#include "OPCDAClientSync.h"
#include <cassert>
#include <iostream>
#include <thread>

using namespace std;

void test (void)
{
	OPCDAClientSync::Initialize();
	OPCDAClientSync opc1 (L"OPC-DA.1");
	for (int i = 0; i < 10000; ++i)
	{
		std::wstring bbb = opc1.ReadItem (L"CKI.OP88G");
		//assert (L"bbb" == bbb);
		wcout<<i<<": "<<bbb<<endl;
		//Sleep (1000);
	}
	OPCDAClientSync::Uninitialize();

}
void test1(void)
{
	OPCDAClientSync::Initialize();
	OPCDAClientSync opc(L"OPC-DA.1");
	for (int i = 0; i < 10000; ++i)
	{
		std::wstring ccc = opc.ReadItem(L"CKI.OP4G");
		//assert (L"bbb" == bbb);
		//wcout << i << ": " << bbb << endl;
		//Sleep (1000);
	}
	OPCDAClientSync::Uninitialize();

}
void main(void)
{
	//test();
	thread testf(test);
	testf.detach();
	thread testd(test1);
	testd.detach();
Sleep (10000);
}`

**Comment by starfisher** [comment]
Did you fix it?

**Comment by starfisher** [comment]
Did you solve it? @h66m9d @shushudu

**Comment by shushudu** [comment]
multithreading not for this app, this is SYNC only
